### PR TITLE
Changes for TarantoolNullField class to singleton

### LINKED ISSUE #195: Make TarantoolNullField class static
Currently we create a new object every time we have a **null** field in a tuple.
Due to performance improvement we need to make behaviour of **TarantoolNullField** class similar to `Optional.empty()`.
Just to have one TarantoolNullField object for all application.

This is a BRAKING CHANGE!

### PR DESCRIPTION
**Pull request by devrishal**
What has been done? Why? What problem is being solved?
The `TarantoolNullField `class was not singleton and its only task it to provide functionality similar to `Optional.Empty()`.
Changes done to make `TarantoolNullField `behaves same as `Optional.Empty()`.

I didn't forget about

- [X] Tests
- [X] Changelog
- [X] Documentation
- [X] Cleanup the code for review. See [checklist](https://github.com/tarantool/cartridge-java/blob/master/docs/review-checklist.md)

Closes #195

### CODE REVIEW

**Review by akudiyar — CHANGES_REQUESTED** [review]
Hi! Thanks for the patch!

I have a couple of questions:

1. I suggest using `create()` as the factory method name. With the `Optional` class we can create instances containing either null or non-null values, but with this class, there is only one possible internal value. So the name `empty()` can imply that something "non-empty" exists too, while `create()` can not.
2. Please fix the incorrect whitespace and run `mvn verify` to fix the other style warnings. 

UPD: discussed the change internally with team, I see that this change may be useful.

**Comment by devrishal** [comment]
@akudiyar Modified the code as per the review comments and also changed for checkstyle.

**Review by ArtDu — APPROVED** [review]
LGTM

**Comment by ArtDu** [comment]
@devrisha I forgot about changelog, can you write one line about changes in CHANGELOG.md?

**Comment by devrishal** [comment]
> @devrisha I forgot about changelog, can you write one line about changes in CHANGELOG.md?

Yes will do now.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # Changelog
 
 ## [Unreleased]
-
+- Changed TarantoolNullField class to singleton
```
Comment (ArtDu) [inline CHANGELOG.md:4]: But you must rebase, because we updated changelog today

Comment (devrishal) [reply CHANGELOG.md]: Please check , did a rebase with the latest changes.

**Review by dkasimovskiy — CHANGES_REQUESTED** [review]

File: src/main/java/io/tarantool/driver/api/tuple/TarantoolNullField.java [hunk 2]
```diff
@@ -12,6 +12,17 @@
  */
 public final class TarantoolNullField implements TarantoolField {
 
+    private static final TarantoolNullField EMPTY = new TarantoolNullField(null);
```
Comment (Elishtar) [inline src/main/java/io/tarantool/driver/api/tuple/TarantoolNullField.java:15]: I suppose we need chose one from two strategy
Not lazy initialization
`public static final TarantoolNullField EMPTY = new TarantoolNullField(null);`
or
lazy initialization
```
private static TarantoolNullField empty;

public static TarantoolNullField create() {
       if (empty == null) 
        empty = new TarantoolNullField(null);;
    }
    
   return empty
}
```

Comment (devrishal) [reply src/main/java/io/tarantool/driver/api/tuple/TarantoolNullField.java:15]: @Elishtar Its already` static final ` and will be immediately instance is assigned to it. The `TarantoolNullField ` is already a final and immutable class. Not sure if you are considering the Thread safety or in any other sense? But to answer the Thread safety the above points can be considered.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,7 +1,7 @@
 # Changelog
 
 ## [Unreleased]
-
+- Changed TarantoolNullField class to singleton ([#195](https://github.com/tarantool/cartridge-java/pull/275))
```
[the remark this answers is not in the document]
Comment (devrishal) [reply CHANGELOG.md:4]: Added new line as per suggestion.

File: src/main/java/io/tarantool/driver/api/tuple/TarantoolNullField.java (hunk 2 above)
Comment (Elishtar) [reply src/main/java/io/tarantool/driver/api/tuple/TarantoolNullField.java:15]: @devrishal if you already initialize  EMPTY, not necessarily use method create for access just change privat to public for EMPTY, if you want use method create. for access you can initialize EMPTY at the moment of first access please chose one strategy, now you use combination

**Review by dkasimovskiy — APPROVED** [review]

File: src/main/java/io/tarantool/driver/api/tuple/TarantoolNullField.java (hunk 2 above)
Comment (devrishal) [reply src/main/java/io/tarantool/driver/api/tuple/TarantoolNullField.java:15]: Please let me know if this is fine for you ```
private static final TarantoolNullField EMPTY = new TarantoolNullField();
    private TarantoolNullField() {
    }

    public static TarantoolNullField create() {
        return EMPTY;
    }
```

**Review by akudiyar — APPROVED** [review]
LGTM.

I agree that lazy init is not needed here, it gives no advantages.

**Comment by akudiyar** [comment]
@devrishal apologies for making that many comments for a simple change, but what do you think about dropping the `create()` method at all and leaving just the public constant? It looks like the method is redundant here since it always returns the constant.
Also, a classic name for such a singleton constant is usually `INSTANCE`.

File: src/main/java/io/tarantool/driver/api/tuple/TarantoolNullField.java (hunk 2 above)
Comment (devrishal) [reply src/main/java/io/tarantool/driver/api/tuple/TarantoolNullField.java]: @akudiyar @Elishtar Please check and let me know if the changes looks good.

**Comment by devrishal** [comment]
> @devrishal apologies for making that many comments for a simple change, but what do you think about dropping the `create()` method at all and leaving just the public constant? It looks like the method is redundant here since it always returns the constant. Also, a classic name for such a singleton constant is usually `INSTANCE`.

No Issues, It is indeed a good learning for me :) . Please let me know if the changes are fine now.

File: src/main/java/io/tarantool/driver/api/tuple/TarantoolNullField.java (hunk 2 above)
Comment (devrishal) [reply src/main/java/io/tarantool/driver/api/tuple/TarantoolNullField.java]: Please check if the changes looks good?

**Review by Elishtar — APPROVED** [review]

**Pull request merged**